packages/changeset-bot/index.test.js

```diff
@@ -0,0 +1,154 @@
+const nock = require('nock');
+const { Probot } = require('probot');
+const changesetBot = require('.');
+const outdent = require('outdent');
+
+const pullRequestOpen = require('./test/fixtures/pull_request.opened');
+const pullRequestSynchronize = require('./test/fixtures/pull_request.synchronize');
+
+nock.disableNetConnect();
+
+describe('changeset-bot', () => {
+  let probot;
+
+  beforeEach(() => {
+    probot = new Probot({});
+    const app = probot.load(changesetBot);
+
+    // just return a test token
+    app.app = () => 'test';
+  });
+
+  it('should add a comment when there is no comment', async () => {
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/issues/1/comments')
+      .reply(200, []);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/files')
+      .reply(200, [
+        { filename: '.changeset/something/changes.md', status: 'added' },
+      ]);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/commits')
+      .reply(200, [{ sha: 'ABCDE' }]);
+
+    nock('https://api.github.com')
+      .post('/repos/pyu/testing-things/issues/1/comments', body => {
+        expect(body.comment_id).toBeNull();
+        return true;
+      })
+      .reply(200);
+
+    await probot.receive({
+      name: 'pull_request',
+      payload: pullRequestOpen,
+    });
+  });
+
+  it('should update a comment when there is a comment', async () => {
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/issues/1/comments')
+      .reply(200, [
+        {
+          id: 7,
+          user: {
+            login: 'changeset-bot[bot]',
+          },
+        },
+      ]);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/files')
+      .reply(200, [
+        { filename: '.changeset/something/changes.md', status: 'added' },
+      ]);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/commits')
+      .reply(200, [{ sha: 'ABCDE' }]);
+
+    nock('https://api.github.com')
+      .patch('/repos/pyu/testing-things/issues/comments/7', body => {
+        expect(body.number).toBe(1);
+        return true;
+      })
+      .reply(200);
+
+    await probot.receive({
+      name: 'pull_request',
+      payload: pullRequestSynchronize,
+    });
+  });
+
+  it('should show correct message if there is a changeset', async () => {
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/issues/1/comments')
+      .reply(200, []);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/files')
+      .reply(200, [
+        { filename: '.changeset/something/changes.md', status: 'added' },
+      ]);
+
+    nock('https://api.github.com')
+      .get('/repos/pyu/testing-things/pulls/1/commits')
+      .reply(200, [{ sha: 'ABCDE' }]);
+
+    nock('https://api.github.com')
+      .post('/repos/pyu/testing-things/issues/1/comments', body => {
```

Comment:
Not a huge fan of `body.body` - Could we spread this out `({ body }) => {` instead?

NB: I have an incredible amount of bias to object spread props and may overuse them. Feel free to change this variable name.